observable.go

```diff
@@ -413,6 +414,74 @@ func (o *observable) Scan(apply Function2) Observable {
 	return &observable{ch: out}
 }
 
+// Compares first items of two sequences and returns true if they are equal and false if
+// they are not. Besides, it returns two new sequences - input sequences without compared items.
+func popAndCompareFirstItems(
+	inputSequence1 []interface{},
+	inputSequence2 []interface{}) (bool, []interface{}, []interface{}) {
+	if len(inputSequence1) > 0 && len(inputSequence2) > 0 {
+		s1, sequence1 := inputSequence1[0], inputSequence1[1:]
+		s2, sequence2 := inputSequence2[0], inputSequence2[1:]
+		return s1 == s2, sequence1, sequence2
+	}
+	return true, inputSequence1, inputSequence2
+}
+
+// SequenceEqual emits true if an Observable and the input Observable emit the same items.
+// When any Observable emits different item, it emits false.
+func (o *observable) SequenceEqual(obs Observable) Observable {
```

Comment:
Since a recent merge in the `v2`, there's another way of creating operators (the reason is for better reuse across different operator types). Your code doesn't compile anymore. Can you please check other operators and follow the same logic?